### ISSUE TITLE
baritone bridging over void in skyblock

### ISSUE THREAD

**Issue by Roboboy5741** [cabaletta/baritone#4288]
just wanted to know how to get the bot to bridge from point A to B in a skyblock world i have modified the throwable items list, made blockplacementpenalty to 0, allowed inventory, enabled assumesafewalk. the bot still has no clue how to get to Point B just walks to the edge of the island and stops

i need the bot to bridge roughly 5-7k blocks with bottom slabs to prevent mob spawning

is there anything im missing or am i doing anything wrong?

**Comment by ZacSharp** [comment]
Not sure whether bottom slabs can be used as scaffolding. Scaffolding is intended to be simple full blocks (dirt, stone, etc.).

**Comment by Roboboy5741** [comment]
> Not sure whether bottom slabs can be used as scaffolding. Scaffolding is intended to be simple full blocks (dirt, stone, etc.).

if its not is there a way to add it and would the bot get confused and place 2 slabs to make full blocks?
in the meantime ill try with normal cobble

(update: got it working with normal cobble but previous question about adding bottomslabs to scaffolding still stands)

**Comment by ZacSharp** [comment]
I don't think it would be too much work, but I also don't see much use in it.
Since you are trying to abuse the pathfinder to effectively build something for you, did you consider using `#selection fill`?

**Comment by Roboboy5741** [comment]
> I don't think it would be too much work, but I also don't see much use in it. Since you are trying to abuse the pathfinder to effectively build something for you, did you consider using `#selection fill`?

would selection fill build a bridge in a diagonal or would it just attempt to make a massive platform

**Comment by ZacSharp** [comment]
Selections are always cuboids so that won't work. Also might be `sel fill` doesn'T allow you to specify specifically _bottom_ slabs.
I think what I'd do is using `buildRepeat` and `buildRepeatCount` to copypaste a handbuilt section of the bridge. So setup repetition settings, build a section of the bridge, select it, copy it, paste it, hope that the builder can actually handle this. (it can place slabs but bridging is harder than that)

**Comment by ZacSharp** [comment]
Screw it, that probably won't work.
What version are you using?

**Comment by Roboboy5741** [comment]
> Screw it, that probably won't work. What version are you using?

1.20.4 also it was funny how fast you turned around and said screw it xD

**Comment by ZacSharp** [comment]
Turns out just removing the safety checks doesn't do the trick :smile:

https://github.com/cabaletta/baritone/assets/68165024/b63bdcb6-76af-4a82-8abb-3620255eb385

Turns out this is a bit harder than I expected. [This version](https://github.com/ZacSharp/baritone/commit/f41c78a03a13b28565c4ae9be909ec49ee777984) ([CI build here](https://github.com/ZacSharp/baritone/actions/runs/8165218525)) can place a line of bottom slabs, but it also comes with at least 4 regressions / new bugs. Some of that might be avoidable by undoing some of the changes (I first tried one approach and then another without undoing changes) but at least 2 of them will stay.
I'm done trying to make slabs usable as throwaways.

**Comment by Roboboy5741** [comment]
listen i seriously appreciate the fact you tried ill take the simple "won't work" and leave it at that sorry for making you jump through hoops xD the bot still helps a ton with getting around my islands through waypoints

(btw just a little bug report of sorts... the bot refuses to walk over my water logged slabs and prefers to try bridging over)
![Screenshot (313)](https://github.com/cabaletta/baritone/assets/57339602/3e4dd644-8589-4f99-8df7-43d6415710c0)
as shown in the picture they go from full block to half block(waterlogged) then back to full block

i might just be dumb and have a setting wrong so call me out on that if i am xD
